app/controllers/application_controller.rb

```diff
@@ -1,2 +1,34 @@
 class ApplicationController < ActionController::Base
+  helper_method :my_blogs, :selected_blog
+
+protected
+  def require_user
+    redirect_to root_path unless current_user
+  end
+
+  def require_blog
+    @blog = Blog.where(id: params[:blog_id] || params[:id]).first
+
+    redirect_to root_path unless @blog
+  end
+
+  def selected_blog
+    Blog.find_by_id(session[:selected_blog_id]) || my_blogs.first
```

Comment:
There intent here was to have to have the user's first blog that is saved in the database.

This might be achieved by changing the `controllers/application_controller.rb` to have
```
def selected_blog
    Blog.find_by_id(session[:selected_blog_id]) || my_blogs.detect(&:persisted?)
  end
```

returning the user's first blog that has actually be persisted.

Reply:
Sounds good to me!

I went with 

```
current_user.blogs.select(&:persisted?)
```

in the `my_blogs` method.